beams/tree_config/action.py

```diff
@@ -27,17 +27,20 @@ def get_tree(self) -> ActionNode:
         def work_func(comp_condition: Evaluatable) -> py_trees.common.Status:
             try:
                 # Set to running
-                value = caget(self.pv)  # double caget, this is uneeded as currently the comp_condition has caget baked in
+                # double caget, this is uneeded as currently the comp_condition
+                # has caget baked in
+                value = caget(self.pv)
+                logger.debug(f" <<-- ({self.name}): caget({self.pv}) -> {value}")
 
                 if comp_condition():
                     return py_trees.common.Status.SUCCESS
-                logger.debug(f"{self.name}: Value is {value}")
 
                 # specific caput logic to SetPVActionItem
                 caput(self.pv, self.value)
+                logger.debug(f" <<-- ({self.name}): caput({self.pv}, {value})")
```

Comment:
This is actually the wrong value, referenced here is the value result from the caget rather than the value we intended to caput.
```suggestion
                logger.debug(f" <<-- ({self.name}): caput({self.pv}, {self.value})")
```

Reply:
ah you're right :shame: